core/network/fabric-interop-cc/contracts/interop/certificate_utils.go

```diff
@@ -112,7 +112,7 @@ func verifyCertificateChain(cert *x509.Certificate, certPEMs []string) error {
 				errMsg := fmt.Sprintf("Certificate link for Subject %s with Parent Subject %s invalid", caCert.Subject.String(), parentCert.Subject.String())
 				return errors.New(errMsg)
 			}
-			if i == len(certPEMs)-1 {
+			if i == len(certPEMs)-1 && cert != nil {
```

Comment:
this check may allow verification to be successful when cert is nil, i.e. that cert is not provided at all...

Also if you want to add it may be just check it when the function starts, so that we can throw error if the `cert` is nil, as we don't want to allow (success verification) if cert is nil. And here it will be assumed that cert is not nil.